libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1312,15 +1312,19 @@ string YulUtilFunctions::allocationFunction()
 	});
 }
 
-string YulUtilFunctions::allocateMemoryArrayFunction(ArrayType const& _type)
+string YulUtilFunctions::allocateMemoryArrayFunction(ArrayType const& _type, bool _shouldZero)
 {
 	solUnimplementedAssert(!_type.isByteArray(), "");
 
 	string functionName = "allocate_memory_array_" + _type.identifier();
```

Comment:
This is something we should really check: The name has to include each and single parameter of the generator function. Otherwise, the second call to this with different parameters will not generate the (different) function.

Reply:
Do you mean I should include the Boolean flag to zero or not as part of it?